src/Net/Http.fs

```diff
@@ -1162,11 +1162,15 @@ module private HttpHelpers =
 #else
             req.ContentLength <- data.Length
 #endif
-            use! output =
-                if Type.GetType("Mono.Runtime") <> null
-                then alternateFromBeginEnd req.BeginGetRequestStream req.EndGetRequestStream req
-                else Async.FromBeginEnd(req.BeginGetRequestStream, req.EndGetRequestStream)
 
+#if FX_NO_LOCAL_FILESYSTEM
+            use! output =
+              if Type.GetType("Mono.Runtime") <> null
+              then alternateFromBeginEnd req.BeginGetRequestStream req.EndGetRequestStream req
+              else Async.FromBeginEnd(req.BeginGetRequestStream, req.EndGetRequestStream)
+#else
+            use! output = req.GetRequestStreamAsync () |> Async.AwaitTask
```

Comment:
added this because it's a framework-built in fix that may work better long-term? can go/no-go with it as you please.